MH-13444, Insecure Series Creation

lkiesow opened the pull request:
This patch prevents users from being able to overwrite series metadata
by default when ingesting event data. Users can still enable this, if
they need this functionality.

In addition to this, Opencast will not any longer add anonymous access
to series if they are created as part of the event ingest.

*Note that the behavior of both features is questionable and we may want
to remove this entirely in the future.*

lkiesow committed:
MH-13444, Insecure Series Creation

This patch prevents users from being able to overwrite series metadata
by default when ingesting event data. Users can still enable this, if
they need this functionality.

In addition to this, Opencast will not any longer add anonymous access
to series if they are created as part of the event ingest.

*Note that the behavior of both features is questionable and we may want
to remove this entirely in the future.*

karendolan commented on modules/ingest-service-impl/src/test/java/org/opencastproject/ingest/impl/IngestServiceImplTest.java:
```diff
@@ -561,7 +561,7 @@ public void testFailedJobs() throws Exception {
 
     // default expectation for series overwrite is True
-    boolean isExpectSeriesOverwrite = true;
+    boolean isExpectSeriesOverwrite = false;
```
Update the comment, too. Or, better, remove the "is True" end portion of it.

karendolan commented on modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java:
```diff
@@ -249,7 +247,7 @@
   /** The default is to overwrite series catalog on ingest */
-  protected boolean defaultIsOverWriteSeries = true;
+  protected boolean defaultIsOverWriteSeries = false;
 
```
Update or remove the old comment.

lkiesow replied:
I went with a static constant I could use here as well

karendolan commented on etc/org.opencastproject.ingest.impl.IngestServiceImpl.cfg:
```diff
@@ -8,7 +8,7 @@
 
```
Update that "default" comment to false.

lkiesow committed:
MH-13444, Make Default Value Constant

…no need for it to be variable at all.

karendolan commented on modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java:
```diff
@@ -880,9 +878,6 @@ protected boolean updateSeries(URI uri) throws IOException, IngestException {
             isUpdated = true;
-            String anonymousRole = securityService.getOrganization().getAnonymousRole();
-            AccessControlList acl = new AccessControlList(new AccessControlEntry(anonymousRole, "read", true));
-            seriesService.updateAccessControl(id, acl);
```
@lkiesow so what default ACL does the new series ultimately get? Is it OC admin read-write-view?  Or can a default be defined in an organizational default series ACL? Our site actually uses a default series ACL when populating a semester's worth of series from a 3rd party.

lkiesow replied:
Not having an ACL, Opencast should fall back to its global default ACL which contains no rules at all. This essentially means that people with exceptions are allowed to access the series. People with exceptions are usually global and organizational administrators.

karendolan commented on modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java:
```diff
@@ -191,6 +191,9 @@
   public static final String ZIP_JOB_LOAD_KEY = "job.load.ingest.zip";
 
+  /** The default is to leave series catalogs untouched on ingest */
+  static final boolean DEFAULT_IS_OVERWRITE_SERIES = false;
```
Ah, that constant, when said out loud, sounds a lot like it's going to overwrite the series ~ "The default is to overwrite the series (false)(??)". It might be better to just to put static final boolean FALSE  = false; :)

lkiesow replied:
```java
  static final boolean DEFAULT_ALLOW_OVERWRITE_SERIES = false;
 ```
?

karendolan replied:
Ah, ok the "IS" version is more programer friendly, just not human friendly.

lkiesow replied:
or maybe 
```java
static final boolean DEFAULT_ALLOW_SERIES_WRITE = false;
```
since we are dealing with writes in general not necessarily only modifications but also series creation.

lkiesow replied:
Updated. Chose yet another name though :D

lkiesow committed:
MH-13444, Better Name For Allowing Series Modifications

…to make clear that the options configure if modifications of existing
series metadata are allowed by event ingests.

lkiesow commented:
@karendolan, can you officially take the review for this pull request and merge it if there are no more issues?
